### PR TITLE
Sourcing necessary compilers and libraries for qhipster in __init__

### PR DESCRIPTION
**Pull request by simonwa7**

### CODE REVIEW

**Review by martamau — COMMENTED** [review]
I think @max-radin had some comments on the source stuff so might be better if he reviews the PR

**Comment by simonwa7** [comment]
After testing with workflow v1, the current method of sourcing the env vars does not work as subprocess uses a new shell (and therefore a different environment). Delaying merge until this is fixed

File: src/python/qeqhipster/simulator.py [hunk 1]
```diff
@@ -3,58 +3,136 @@
...
+
 class QHipsterSimulator(QuantumSimulator):
     def __init__(self, n_samples=None, nthreads=1):
         self.nthreads = nthreads
         self.n_samples = n_samples
 
+        # NOTE: The environment variables that are set below are necessary for running qhipster with the intel psxe
+        #   runtime installation. They were obtained through sourcing the script
+        #   /app/usr/local/bin/compilers_and_library.sh which can be found in the zapatacomputing/qe-qhipster docker
+        #   image.
```
Comment (mstechly) [inline src/python/qeqhipster/simulator.py:25]: For me it's an unnecessary line break ;)

**Review by mstechly — APPROVED** [review]
I don't get what's happening in the `os.putenv` part, but apart from that looks good :)

**Pull request merged**